### PR TITLE
docs: Add HTTP API swagger documentation

### PR DESCRIPTION
**Pull request by netoax**
What does this implement/fix? Explain your changes
---------------------------------------------------
Adds the necessary swagger tooling and generates the current HTTP endpoints
documentation by using it. So, when running the `make http-docs` command
the endpoint annotations will be parsed to generate the static files that
will be served via `/swagger/index.html` endpoint.

Does this close any currently open issues?
------------------------------------------
This PR will get the #10 documentation task partially done.

Where has this been tested?
---------------------------

**Operating System/Platform:**  macOS 10.14 - Darwin 18.0.0

**Go Version:** go1.12.7 darwin/amd64

### CODE REVIEW

**Comment by netoax** [comment]
I just forgot to add the appropriate instructions to the README. I'll fix this later.

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/knot-babeltower/knot-babeltower/pullRequest?prid=4615084&bid=15479225)

File: pkg/server/server.go [hunk 1]
```diff
@@ -5,9 +5,11 @@ import (
 	"fmt"
 	"net/http"
 
+	httpSwagger "github.com/swaggo/http-swagger"
+
+	_ "github.com/CESARBR/knot-babeltower/docs"
```
Comment (joaoaneto) [inline pkg/server/server.go]: Codacy Issue found: [a blank import should be only in a main or test package, or have a comment justifying it](https://app.codacy.com/app/joaoaneto/knot-babeltower/pullRequest?prid=4615084)

File: docs/docs.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        "server.Health": {
+            "type": "object",
+            "properties": {
+                "status": {
+                    "type": "string"
+                }
+            }
+        }
+    }
+}`
```
Comment (ramonhpr) [inline docs/docs.go]: Is it common to add a JSON like that rather than a file?

File: docs/swagger.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      - description: User e-mail and password
+        in: body
+        name: user
+        required: true
+        schema:
+          $ref: '#/definitions/entities.User'
+      produces:
+      - application/json
+      responses:
+        "201":
```
Comment (ramonhpr) [inline docs/swagger.yaml:54]: Add fail responses too

File: docs/swagger.json [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+                        "name": "user",
+                        "in": "body",
+                        "required": true,
+                        "schema": {
+                            "$ref": "#/definitions/entities.User"
+                        }
+                    }
+                ],
+                "responses": {
+                    "201": {
```
Comment (ramonhpr) [inline docs/swagger.json:53]: Add fail responses too

**Review by ramonhpr — CHANGES_REQUESTED** [review]

File: docs/docs.go (hunk 2 above)
Comment (netoax) [reply docs/docs.go]: This file is generated by the swagger documentation tool.

File: docs/swagger.yaml (hunk 3 above)
Comment (netoax) [reply docs/swagger.yaml:54]: For the `/healthcheck` endpoint, I don't know if I need to map a specific error.

File: docs/swagger.json (hunk 4 above)
Comment (netoax) [reply docs/swagger.json:53]: Done.

File: README.md [hunk 5]
```diff
@@ -58,3 +58,13 @@ A container is specified at `docker/Dockerfile`. To use it, execute the followin
 ```bash
 curl http://<hostname>:<port>/healthcheck
 ```
+
+### Documentation
+
+Server documentation is auto-generated by the `swag` tool (<https://github.com/swaggo/swag>) from annotations placed in the code and can be viewed on the browser:
+
+```
```
Comment (joaoaneto) [inline README.md]: Codacy Issue found: [[fenced-code-flag] Missing code-language flag](https://app.codacy.com/app/joaoaneto/knot-babeltower/pullRequest?prid=4615084)

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/joaoaneto/knot-babeltower/pullRequest?prid=4615084&bid=15479225)

**Review by ramonhpr — APPROVED** [review]

**Pull request merged**